deployments/scripts/storage/rustfs.sh

```
@@ -0,0 +1,121 @@
+#    to match how OSMO talks to the in-cluster endpoint (a bare Service DNS name
+#    isn't virtual-host addressable). head-bucket makes this idempotent without
+#    parsing provider-specific "already exists" error strings. A unique
+#    per-invocation pod name avoids collisions with a prior run's helper pod
+#    stuck Terminating; `--rm` reaps it; `timeout` guards against stuck image
+#    pull / Pending-forever scheduling.
+BUCKET_SETUP_TIMEOUT="${BUCKET_SETUP_TIMEOUT:-300}"
+AWS_CLI_IMAGE="${AWS_CLI_IMAGE:-amazon/aws-cli:latest}"
+BUCKET_SETUP_POD="rustfs-bucket-setup-$RANDOM-$RANDOM"
+echo "[INFO] Ensuring RustFS bucket $RUSTFS_BUCKET exists (helper pod: $BUCKET_SETUP_POD)"
+timeout "$BUCKET_SETUP_TIMEOUT" \
+  $KUBECTL run "$BUCKET_SETUP_POD" --rm -i --restart=Never \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)